### PR TITLE
Add highlight delay

### PR DESCRIPTION
**Pull request by heathkit**
Also, log whenever we wait for Angular.

### CODE REVIEW

File: lib/angular_wait_barrier.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import {SimpleWebDriverClient} from './simple_webdriver_client';
+import {WebDriverCommand} from './webdriver_commands';
+import {WebDriverLogger} from './webdriver_logger';
+import {WebDriverBarrier} from './webdriver_proxy';
+
+let angularWaits = require('./client_scripts/wait.js');
+
+export class AngularWaitBarrer implements WebDriverBarrier {
```
Comment (cnishina) [inline lib/angular_wait_barrier.ts]: Should have a JSDoc comment block about the `AngularWaitBarrer` class. Also, is it supposed to be `AngularWaitBarrer` or `AngularWaitBarrier` (spellling?)

File: lib/angular_wait_barrier.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  rootSelector: string;
+  enabled: boolean;
+  logger: WebDriverLogger;
+
+  constructor(private client: SimpleWebDriverClient) {
+    this.enabled = true;
+    this.rootSelector = '';
+  }
+
+  setRootSelector(selector: string) {
```
Comment (cnishina) [inline lib/angular_wait_barrier.ts:37]: Should have a JSDoc comment block when `setRootSelector` how it is used or what it is doing.

File: lib/angular_wait_barrier.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    //
+    // See https://code.google.com/p/selenium/wiki/JsonWireProtocol for
+    // descriptions of the paths.
+    // We shouldn't stabilize if we haven't loaded the page yet.
+    let parts = url.split('/');
+    if (parts.length < 4) {
+      return false;
+    }
+
+    let commandsToWaitFor = [
```
Comment (cnishina) [inline lib/angular_wait_barrier.ts]: nit: should use `const` over `let`?

File: lib/simple_webdriver_client.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import * as http from 'http';
+import * as url from 'url';
+
+export class SimpleWebDriverClient {
```
Comment (cnishina) [inline lib/simple_webdriver_client.ts:8]: Should have a JSDoc comment block about `SimpleWebDriverClient`. Might also want to consider writing comment blocks over other public methods.

**Review by cnishina — CHANGES_REQUESTED** [review]

File: lib/angular_wait_barrier.ts (hunk 1 above)
Comment (heathkit) [reply lib/angular_wait_barrier.ts]: Done

File: lib/angular_wait_barrier.ts (hunk 2 above)
Comment (heathkit) [reply lib/angular_wait_barrier.ts:37]: Done

File: lib/angular_wait_barrier.ts (hunk 3 above)
Comment (heathkit) [reply lib/angular_wait_barrier.ts]: Yeah, I should have a lint for that.

File: lib/simple_webdriver_client.ts (hunk 4 above)
Comment (heathkit) [reply lib/simple_webdriver_client.ts:8]: Done

File: lib/simple_webdriver_client.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          let response = JSON.parse(respData);
+          if (response.state !== 'success') {
+            reject(response.value);
+          }
+          resolve(response.value);
+        });
+      });
+    });
+  };
+}
```
Comment (juliemr) [inline lib/simple_webdriver_client.ts]: nit: add newline at eof

File: lib/client_scripts/highlight.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a floating transparent div at the specified location in order to highlight a particular
+ * element. Note that these scripts are run directly in the browser under test, so they need to
+ * be ES5 compatible and serializable.
+ */
+exports.HIGHLIGHT_FN = function(top, left, width, height) {
+  console.log('Highlighting at ', top, left, width, height);
+  var el = document.createElement('div');
+  el.id = 'bpElementHighlight';
```
Comment (juliemr) [inline lib/client_scripts/highlight.js]: Can this be uglier and have some more `___`s in it?

File: lib/client_scripts/highlight.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a floating transparent div at the specified location in order to highlight a particular
```
Comment (juliemr) [inline lib/client_scripts/highlight.js]: Nit: s/transparent/translucent

**Review by juliemr — APPROVED** [review]

File: lib/client_scripts/highlight.js (hunk 7 above)
Comment (heathkit) [reply lib/client_scripts/highlight.js]: done

File: lib/simple_webdriver_client.ts (hunk 5 above)
Comment (heathkit) [reply lib/simple_webdriver_client.ts]: done

**Pull request merged**